common/src/main/java/io/pravega/common/health/processor/impl/NettyOutputStream.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.health.processor.impl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Class that wraps a netty connection in an outputstream.
+ * This class is used to write the health reports.
+ */
+public class NettyOutputStream extends OutputStream {
+    private final Channel channel;
+
+    public NettyOutputStream(Channel channel) {
+        this.channel = channel;
+    }
+
+    @Override
+    public void close() throws IOException {
+        channel.close();
+    }
+
+    @Override
+    public void flush() throws IOException {
+        channel.flush();
```

Comment:
Netty's flush works differently than outputstream's flush. You never need to call flush on output stream, it's just a way to force data to be written. Netty will NEVER write any data over the wire until flush is called.